personal_finances/user/user_service.py

```diff
@@ -0,0 +1,93 @@
+import json
+import logging
+import os
+from typing import Any, Callable
+from decorator import decorator
+from personal_finances.user.user_login import user_login
+from personal_finances.user.user_auth_exceptions import (
+    InvalidLambdaEventInput,
+    AuthorizationHeaderNotPresent,
+    AuthorizationHeaderEmptyContent,
+    get_server_response_by_exception,
+)
+from personal_finances.user.user_persistence import get_user_ibans, update_user_iban
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_auth_header(event: dict) -> str:
+    headers = event.get("headers", None)
+    if headers is None:
+        raise InvalidLambdaEventInput()
+
+    auth_header = headers.get("Authorization", None)
+    if auth_header is None:
+        raise AuthorizationHeaderNotPresent()
+
+    if auth_header == "":
+        raise AuthorizationHeaderEmptyContent()
+
+    return str(auth_header)
+
+
+def get_allowed_domain() -> str:
+    # TODO: allow for multiple origins
+    # fetching only the first origin
+    try:
+        allowed_domain = os.environ["ALLOWED_ORIGIN_DOMAINS"].split(",")[0]
+    except Exception as e:
+        LOGGER.info(f"falling 'ALLOWED_ORIGIN_DOMAINS' back to * due to: {e}")
+        return "*"
+    return allowed_domain if allowed_domain else "*"
+
+
+def add_cors_to_dict(input_dict: dict) -> dict:
+    return {
+        **input_dict,
+        "headers": {"Access-Control-Allow-Origin": get_allowed_domain()},
+    }
+
+
+@decorator
+def add_cors_decorator(handler: Callable, *args: Any, **kwargs: Any) -> dict:
+    return add_cors_to_dict(handler(*args, **kwargs))
+
+
+@add_cors_decorator
+def user_service_handler(event: dict, context: str) -> dict:
+    resource = event["resource"]
+    http_method = event["httpMethod"]
+
+    logging.info(f"info_event:{event}")
+    logging.info(f"info_context:{context}")
+
+    try:
+        if resource == "/users/{userId}/login" and http_method == "POST":
+            auth_header = _get_auth_header(event)
+            token_json = user_login(auth_header)
+            return {"statusCode": 200, "body": f"{token_json}"}
+
+        elif resource == "/users/{userId}/bank-accounts" and http_method == "GET":
+            user_id = event["pathParameters"].get("userId")
+            iban_list = get_user_ibans(user_id)
+            return {"statusCode": 200, "body": f"ibanList:{json.dumps(iban_list)}"}
+
+        elif (
+            resource == "/users/{userId}/bank-accounts/{iban}" and http_method == "PUT"
+        ):
+            iban = event["pathParameters"].get("iban")
+            user_id = event["pathParameters"].get("userId")
+            update_user_iban(user_id, iban)
+            return {"statusCode": 200, "body": "IBAN added successfully"}
```

Comment:
Change this to a router map:
```
router_map = {
  'some/api/path': some_handler_function_for_that_path,
  'some/api/path1': some_handler_function_for_that_path1,
  'some/api/path2': some_handler_function_for_that_path2
}
```

Reply:
Issue created for that:
Create a router map for the user service api #57